FirebaseDynamicLinks/Sources/FIRDynamicLinks.m

```diff
@@ -417,17 +417,37 @@ - (nullable FIRDynamicLink *)dynamicLinkFromCustomSchemeURL:(NSURL *)url {
         NSString *urlString = parameters[kFIRDLParameterLink];
         NSURL *deepLinkURL = [NSURL URLWithString:urlString];
         if (deepLinkURL) {
-          FIRDynamicLink *dynamicLink = [[FIRDynamicLink alloc] initWithParametersDictionary:@{
-            kFIRDLParameterDeepLinkIdentifier : urlString,
-            kFIRDLParameterCampaign : parameters[kFIRDLParameterCampaign] ?: [NSNull null],
-            kFIRDLParameterContent : parameters[kFIRDLParameterContent] ?: [NSNull null],
-            kFIRDLParameterMedium : parameters[kFIRDLParameterMedium] ?: [NSNull null],
-            kFIRDLParameterSource : parameters[kFIRDLParameterSource] ?: [NSNull null],
-            kFIRDLParameterTerm : parameters[kFIRDLParameterTerm] ?: [NSNull null],
-            kFIRDLParameterMinimumAppVersion : parameters[kFIRDLParameterMinimumAppVersion]
-                ?: [NSNull null],
-          }];
+          NSMutableDictionary *paramsDictionary = [[NSMutableDictionary alloc]
+              initWithDictionary:@{kFIRDLParameterDeepLinkIdentifier : urlString}];
+
+          if (parameters[kFIRDLParameterSource] != nil) {
+            [paramsDictionary setValue:parameters[kFIRDLParameterSource]
+                                forKey:kFIRDLParameterSource];
+          }
+
+          if (parameters[kFIRDLParameterMedium] != nil) {
+            [paramsDictionary setValue:parameters[kFIRDLParameterMedium]
+                                forKey:kFIRDLParameterMedium];
+          }
+
+          if (parameters[kFIRDLParameterTerm] != nil) {
+            [paramsDictionary setValue:parameters[kFIRDLParameterTerm] forKey:kFIRDLParameterTerm];
+          }
+
+          if (parameters[kFIRDLParameterCampaign] != nil) {
+            [paramsDictionary setValue:parameters[kFIRDLParameterCampaign]
+                                forKey:(kFIRDLParameterCampaign)];
+          }
+
+          if (parameters[kFIRDLParameterContent] != nil) {
+            [paramsDictionary setValue:parameters[kFIRDLParameterContent]
+                                forKey:kFIRDLParameterContent];
+          }
+
+          FIRDynamicLink *dynamicLink =
+              [[FIRDynamicLink alloc] initWithParametersDictionary:paramsDictionary];
           dynamicLink.matchType = FIRDLMatchTypeUnique;
+          dynamicLink.minimumAppVersion = parameters[kFIRDLParameterMinimumAppVersion];
```

Comment:
Why doesn't this have the same `nil` checking? It looks like it can be nil from its documentation: `@param minAppVersion The minimum app version string, if any, found in the response from a server`

Reply:
Yeah thats true. 

But it seems like @cavalos0086 reverted it back to the old code: https://github.com/firebase/firebase-ios-sdk/pull/10413/files#diff-38f8dcf2a819794228e1caf6b2b19c7432ceadfbd049fb77437a86c08b47cd09L423